File: packages/targets/test/goldens/Bindings/diez-stub-android/src/main/java/org/diez/stub/Bindings.kt
package org.diez.stub

data class Bindings(
    val image: Image = Image(File("assets/image%20with%20spaces.jpg", "image"), File("assets/image%20with%20spaces@2x.jpg", "image"), File("assets/image%20with%20spaces@3x.jpg", "image"), File("assets/image%20with%20spaces@4x.jpg", "image"), Size2D(246F, 246F)),
    val lottie: Lottie = Lottie(File("assets/lottie.json", "raw"), true, true),
    val typograph: Typograph = Typograph(Font(File("assets/SomeFont.ttf", "font"), "SomeFont"), 50F, Color(0.16666666666666666F, 1F, 0.5F, 1F), true),
    val linearGradient: LinearGradient = LinearGradient(arrayOf<GradientStop>(GradientStop(0F, Color(0F, 1F, 0.5F, 1F)), GradientStop(1F, Color(0.6666666666666666F, 1F, 0.5F, 1F))), Point2D(0F, 0.5F), Point2D(1F, 0.5F)),
    val point: Point2D = Point2D(0.5F, 0.5F),
    val size: Size2D = Size2D(400F, 300F),
    val shadow: DropShadow = DropShadow(Point2D(1F, 2F), 3F, Color(0.3333333333333333F, 1F, 0.5F, 0.5F))
) : StateBag {
    companion object {}
    override val name = "Bindings"
}

File: packages/stdlib/test/fixtures/Bindings/Bindings.ts
import {
  Color,
  DropShadow,
  File,
  FileType,
  Fill,
  Font,
  Image,
  LinearGradient,
  Lottie,
  Panel,
  Point2D,
  Size2D,
  Toward,
  Typograph,
} from '@diez/prefabs';

export class Bindings {
  image = Image.responsive('assets/image with spaces.jpg', 246, 246);

  lottie = Lottie.fromJson('assets/lottie.json');

  typograph = new Typograph({
    font: new Font({
      name: 'SomeFont',
      file: new File({src: 'assets/SomeFont.ttf', type: FileType.Font}),
      weight: 700,
      fallbacks: ['Verdana', 'serif'],
    }),
    fontSize: 50,
    color: Color.hex('#ff0'),
  });

  linearGradient = LinearGradient.make(Toward.Right, Color.rgb(255, 0, 0), Color.rgb(0, 0, 255));

  point = Point2D.make(0.5, 0.5);

  size = Size2D.make(400, 300);

  shadow = new DropShadow({
    offset: Point2D.make(1, 2),
    radius: 3,
    color: Color.rgba(0, 255, 0, 0.5),
  });

  fill = Fill.color(Color.rgb(255, 0, 0));

  panel = new Panel({
    cornerRadius: 5,
    background: Fill.color(Color.rgb(0, 0, 255)),
    dropShadow: new DropShadow({
      offset: Point2D.make(2, 3),
      radius: 4,
      color: Color.rgb(255, 0, 0),
    }),
    elevation: 6,
  });

  color = new Color();

  file = new File({src: 'assets/SomeFile.txt'});
}
